File: Consulting/Enrollment/src/gfu_predict.c
#include <stdio.h>
#include <string.h>
#include "doublefann.h"

#define NUM_ARGS 3
#define MAX_STRING 1024


int main(int argc, char* argv[])
{
   int i;
   struct fann_train_data *data;
   struct fann *network;
   fann_type prediction;

   // Sufficient Arguments?
   if (argc != NUM_ARGS)
   {
      fprintf(stderr, "Insufficient arguments\nUsage: %s <network file> <data file>\n", argv[0]);
      exit(-1);
   }

   
   // Open the network file 
   if ( !(network = fann_create_from_file(argv[1])) )
   {
      perror("Error creating network --- ABORTING.\n");
      return -1;
   }

   // Open the data file
   if ( !(data = fann_read_train_from_file(argv[2])) )
   {
      fprintf(stderr, "Error opening data file %s\n", argv[2]);
      return -1;
   }

   // Iterate over every entry in the datafile and make a prediction
   for (i=0; i<fann_length_train_data(data); i++)
   {
      fann_scale_input(network, data->input[i]);
      prediction = fann_run(network, data->input[i])[0];
      printf("%22.20lf\n", prediction);
   }

   return EXIT_SUCCESS;
}

File: Consulting/Enrollment/src/gfu_predict.py
import fann2.libfann as fann
import sys


_NUM_ARGS = 3

'''
print "BUGGER - does not scale input"
sys.exit(-1)
'''

if __name__=='__main__':
    # Sufficient Arguments?
    if (len(sys.argv) != _NUM_ARGS):
        print 'Insufficient arguments\nUsage: ',sys.argv[0],' <network file> <data file>\n'
        sys.exit(-1)

    network = fann.neural_net()
    network.create_from_file(sys.argv[1])

    # Open and read the file of data to predict
    data = fann.training_data()
    data.read_train_from_file(sys.argv[2])
    network.scale_train(data)

    # Iterate over every entry in the datafile and make a prediction
    for anInput in data.get_input():
        print network.run(anInput)[0]
